app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt

```diff
@@ -258,3 +274,79 @@ fun DisplayUserPicture(user: User?, size: Dp, testTag: String) {
                 .testTag(testTag))
   }
 }
+
+/**
+ * This function displays the friend menu, which contains actions to perform on a friend.
+ *
+ * @param showMenu - The state of the menu (expanded or not).
+ * @param friend - The friend to perform actions on.
+ * @param userViewModel - The view model for the user / the friend.
+ * @param context - The context of the application.
+ */
+@Composable
+fun FriendMenu(
+    showMenu: MutableState<Boolean>,
+    friend: User,
+    userViewModel: UserViewModel,
+    context: Context
+) {
+  DropdownMenu(
+      expanded = showMenu.value,
+      onDismissRequest = { showMenu.value = false },
+      modifier = Modifier.testTag("friendMenu")) {
+        val showConfirmDialog = remember { mutableStateOf(false) }
+        var onConfirm = {}
+        var onDismiss = {}
+
+        CustomDialog(
+            showDialog = showConfirmDialog,
+            dialogType = DialogType.CONFIRM,
+            tag = "RemoveFriend",
+            title = context.getString(R.string.RemoveFriendTitle),
+            verbose = false,
+            onSubmit = { onConfirm() },
+            onDismiss = { onDismiss() },
+            Content = ({ Text(context.getString(R.string.RemoveFriendContent, friend.username)) }))
+
+        DropdownMenuItem(
+            modifier = Modifier.testTag("RemoveFriendMenuItem"),
+            onClick = {
+              val friendUID = friend.uid
+              val friendName = friend.username
+
+              val currentUID = userViewModel.currentUser.value?.uid
+
+              if (currentUID != null) {
+                // Set up the confirm function for the dialog
+                Log.d("Profile", "Setting up callback (confirm) for removing friend")
```

Comment:
It would be a good idea to store the “_Profile_” log tag in a const variable and use it in the rest of the file. What do you think of it?

Reply:
I agree that it can prevents typos, which would lead to logs not appearing / multiple profile tags.